modules/sdk-coin-ada/src/lib/transactionBuilder.ts

```diff
@@ -102,21 +122,32 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
         )
       );
     });
-    const outputs = CardanoWasm.TransactionOutputs.new();
+    let outputs = CardanoWasm.TransactionOutputs.new();
+    let totalAmountToSend = CardanoWasm.BigNum.zero();
     this._transactionOutputs.forEach((output) => {
+      const amount = CardanoWasm.BigNum.from_str(output.amount);
       outputs.add(
         CardanoWasm.TransactionOutput.new(
           CardanoWasm.Address.from_bech32(output.address),
-          CardanoWasm.Value.new(CardanoWasm.BigNum.from_str(output.amount))
+          CardanoWasm.Value.new(amount)
         )
       );
+      totalAmountToSend = totalAmountToSend.checked_add(amount);
     });
+
+    // add extra output for the change
+    if (this._changeAddress) {
+      const changeAddress = CardanoWasm.Address.from_bech32(this._changeAddress);
+      outputs.add(CardanoWasm.TransactionOutput.new(changeAddress, CardanoWasm.Value.new(CardanoWasm.BigNum.zero())));
```

Comment:
why the amount is zero for change address?

Reply:
You start off with making a dummy change address output with zero as a placeholder, then, we calculate the fee. Once we have the fee we can calculate the change value for real

Reply:
will 0 affect the final size calculation?